### ISSUE TITLE
Python: Support tumbling window based on punctuation

### ISSUE THREAD

**Issue by markheger** [IBMStreams/streamsx.topology#2527]
similar to SPL code:
```
   Agg1 = Aggregate(Beat)                                                        
    {                                                                                 
      window                                                                          
        Beat : tumbling, punct();  
```

**Comment by markheger** [comment]
merged to branch feature/punct

**Comment by markheger** [comment]
create a tumbling window with eviction by punctuation:
`window = s.batch('punct')`

**Comment by markheger** [comment]
Python window.Aggregate() needs to support a "**_punctWindow_**".

Affected operator:
`com.ibm.streamsx.topology/com.ibm.streamsx.topology.functional.python/Aggregate/Aggregate_cpp.cgt`

**Comment by markheger** [comment]
```
        window = s.batch('punct')
        a = window.aggregate(SumMaxValues())
```

**Comment by markheger** [comment]
merged to develop branch for 1.16.0a
